src/dotnet/ToolPackage/ToolPackageInstaller.cs

```diff
@@ -161,7 +161,17 @@ private string JoinSourceAndOfflineCache(string[] additionalFeeds)
             var feeds = new List<string>();
             if (additionalFeeds != null)
             {
-                feeds.AddRange(additionalFeeds);
+                foreach (var feed in additionalFeeds)
+                {
+                    if (Uri.IsWellFormedUriString(feed, UriKind.Absolute))
+                    {
+                        feeds.Add(feed);
+                    }
+                    else
+                    {
+                        feeds.Add(Path.GetFullPath(feed));
```

Comment:
The wacky relative path on Windows is drive relative (`C:foo`). It looks like that one falls through to here, which is good. Device paths are strange too (`\\?\` & `\\.\`), but they always have to be fully qualified.

There is a new `Path.GetFullPath` that takes the base path to resolve relative paths against. It hardens against the current directory getting changed by other process threads. Consider that if you know what the base path should be.

Reply:
The base path should be CWD in this case (at least, the CWD of `dotnet` when it was created).  `dotnet` only modifies the CWD from a legacy `dotnet migrate` scenario, as far as I know, and that doesn't impact this scenario at all.